pkg/config/config.go

```diff
@@ -0,0 +1,333 @@
+package config
+
+import (
+	"github.com/golang/protobuf/ptypes/any"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/net/context"
+
+	api "github.com/osrg/gobgp/api"
+	"github.com/osrg/gobgp/internal/pkg/apiutil"
+	"github.com/osrg/gobgp/internal/pkg/config"
+	"github.com/osrg/gobgp/internal/pkg/table"
+	"github.com/osrg/gobgp/pkg/packet/bgp"
+	"github.com/osrg/gobgp/pkg/server"
+)
+
+func ReadConfigFile(configFile, configType string) (*config.BgpConfigSet, error) {
+	return config.ReadConfigfile(configFile, configType)
+}
+
+func ReadConfigFileOnSighup(configFile, configType string) chan *config.BgpConfigSet {
+	ch := make(chan *config.BgpConfigSet)
```

Comment:
Returning a channel is a preferable design of a public API? For example, who is responsible for closing?

Reply:
That is a fair point. I had a difficult time deciding where to draw the line here, especially given how [this code][1] was being used by the `cmd/gobgpd/main.go` file [here][2] and [here][3]. However,, now that I look at this, the last commit in this PR refactors [it][1] to expose the `ReadConfigFile` method which is friendlier. Let me see if I can clean this up with on more commit...

[1]: https://github.com/osrg/gobgp/blob/2f5a4d59cb8d494bcf1a64d45d84bcd8fd4a7f06/internal/pkg/config/serve.go#L27-L76
[2]: https://github.com/osrg/gobgp/blob/2f5a4d59cb8d494bcf1a64d45d84bcd8fd4a7f06/cmd/gobgpd/main.go#L187
[3]: https://github.com/osrg/gobgp/blob/2f5a4d59cb8d494bcf1a64d45d84bcd8fd4a7f06/cmd/gobgpd/main.go#L187